hazelcast/src/main/java/com/hazelcast/core/PartitioningStrategy.java

```diff
@@ -16,13 +16,15 @@
 
 package com.hazelcast.core;
 
+import java.io.Serializable;
+
 /**
  * PartitioningStrategy allows keys to be located on the same member.
  * This causes related data to be stored in the same location. (See data-affinity.)
  *
  * @param <K> key type
  */
-public interface PartitioningStrategy<K> {
+public interface PartitioningStrategy<K> extends Serializable {
```

Comment:
After talking with @toktarev it is fine to be serializable.  Values won't be send very often.